proofs/tree.go

```diff
@@ -386,13 +398,17 @@ func getIndexOfString(slice []string, match string) (index int, err error) {
 
 // HashTwoValues concatenate two hashes to calculate hash out of the result. This is used in the merkleTree calculation code
 // as well as the validation code.
-func HashTwoValues(a, b []byte) (hash []byte) {
+func HashTwoValues(a []byte, b []byte, hashFunc hash.Hash) (hash []byte) {
 	data := make([]byte, 64)
 	copy(data[:32], a[:32])
 	copy(data[32:], b[:32])
-	h := blake2b.Sum256(data)
-	hash = h[:]
-	return
+
+	defer hashFunc.Reset()
```

Comment:
As we use hashing in a couple of places, I would wrap this into its own function called `performHashing` or similar. So it is reusable.